great_tables/_tbl_data.py

```diff
@@ -207,27 +246,46 @@ def _set_cell(data: DataFrameLike, row: int, column: str, value: Any):
 
 
 @_set_cell.register(PdDataFrame)
-def _(data, row: int, column: str, value: Any) -> None:
+def _(data, row: int, column: str, value: Any) -> PdDataFrame:
     # TODO: This assumes column names are unique
     # if this is violated, get_loc will return a mask
     col_indx = data.columns.get_loc(column)
     data.iloc[row, col_indx] = value
+    return data
 
 
 @_set_cell.register(PlDataFrame)
-def _(data, row: int, column: str, value: Any) -> None:
+def _(data, row: int, column: str, value: Any) -> PlDataFrame:
     data[row, column] = value
+    return data
```

Comment:
Does they PyArrowTable version return a copy of the data? For polars and pandas, I think returning None is useful for [command-query separation](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation) (where `_set_cell` is a command, so shouldn't return a value).

But it seems totally fair if PyArrowTable can't  be mutated. (I'm guessing at some point we'll want to change `_set_cell` to set entire columns, anyway?!)

Reply:
Correct, in PyArrow data is immutable, so it is not possible to mutate in-place the Table (at least from public APIs point of view) and by design of Arrow it is considered bad practice to mutate it in general.

I understand the point of command-query separation, my approach has been to retain as much as possible of the existing polars/pandas implementation of the functions (to focus PR changes to Arrow), and thus the implementations ended up mixing the two behaviours between the various backends, but I think that the cleanest approach would be to move all of them to return a copy instead of mutating it in-place. That seems to be more future-proof as it doesn't constraint the codebase to only work with mutable structures.

Reply:
PS: Given that the scope of great-tables is rendering tables for human consumption, we should be talking about relatively small data, so I think that even involving `Dataframe.copy` on pandas should introduce an acceptable cost

Reply:
https://github.com/posit-dev/great-tables/pull/487/commits/d40040b00d5738ae09e42c1ba71b28f6ca5efea0 consolidates the behavior of the functions